### PR TITLE
REF Use the new prepareParamsForPaymentProcessor function in more places

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Follow on from #15280

Before
----------------------------------------
More code duplication

After
----------------------------------------
More code consolidation

Technical Details
----------------------------------------

Comments
----------------------------------------
#15280 should be reviewed and merged first.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire  makes sense - wanna rebase it

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -1228,9 +1228,7 @@ public function submit($params) {
 
     if ($this->_mode) {
       // add all the additional payment params we need
-      $this->_params["state_province-{$this->_bltID}"] = $this->_params["billing_state_province-{$this->_bltID}"] = CRM_Core_PseudoConstant::stateProvinceAbbreviation($this->_params["billing_state_province_id-{$this->_bltID}"]);
-      $this->_params["country-{$this->_bltID}"] = $this->_params["billing_country-{$this->_bltID}"] = CRM_Core_PseudoConstant::countryIsoCode($this->_params["billing_country_id-{$this->_bltID}"]);
-
+      $this->_params = $this->prepareParamsForPaymentProcessor($this->_params);
```
Comment (stesi561) [inline CRM/Event/Form/Participant.php]: Does it matter that now these are only being set if `!empty($params["billing_state_province_id-{$this->_bltID}"])` and similarly for country id?
 
and that
$params['state_province']
and 
 $params['country']

are now being set as well? I assume not but just wanted to double check.

Comment (stesi561) [reply CRM/Event/Form/Participant.php]: Or is the fact that we were not previous testing for empty values the problem!

**Comment by stesi561** [comment]
The only thing I could spot is that I think there is a slight change in behaviour here - that I'm not sure is important or not. I've commented on the file concerned.

* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) PASS - perhaps more description would be good but the title sums it up.
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)):  PASS  
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) N/A
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) Test build has expired. 
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) see above.
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) PASS
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) N/A
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) PASS

**Comment by stesi561** [comment]
To r-run this do I need to set up an actual payment processor or will the dummy payment process give enough coverage?

File: CRM/Event/Form/Participant.php (hunk 1 above)
Comment (mattwire) [reply CRM/Event/Form/Participant.php]: @stesi561 The fact we weren't checking for empty values would potentially cause PHP notices.  And missing params that are now being set means we might now be saving data that might not have been saved previously (or the payment processor had to implement custom handling to look for the parameter) - eg. for Country.

**Comment by mattwire** [comment]
@stesi561 It should be sufficient to test with the dummy payment processor.  Really we are just checking that there is no loss of information being saved and no new PHP notices.

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
@stesi561 Thankyou for reviewing / testing.  I've triggered the test build again if you have time for r-run?

**Comment by stesi561** [comment]
My initial test of http://core-15281-8xhkx.test-1.civicrm.org:8001/civicrm/event/register?reset=1&id=1 while logged in resulted in the below error - this happened before the confirmation page however I can't seem to replicate this again - as either logged in or anon. This error seems to be related to a missing id in the url - the changes don't seem to relate to this so I don't think this is cause for concern.

No warnings/notices showing in watchdog/or on screen.

Address data appears to be saving correctly.

```
#0 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Core/Error.php(381): CRM_Core_Error::backtrace()
#1 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Utils/Request.php(121): CRM_Core_Error::fatal("Could not find valid value for id")
#2 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Event/StateMachine/Registration.php(52): CRM_Utils_Request::retrieve("id", "Positive", Object(CRM_Event_Controller_Registration), TRUE)
#3 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Event/Controller/Registration.php(47): CRM_Event_StateMachine_Registration->__construct(Object(CRM_Event_Controller_Registration), TRUE)
#4 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Core/Invoke.php(285): CRM_Event_Controller_Registration->__construct("Event Registration", TRUE, "null", NULL, "false")
#5 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Core/Invoke.php(84): CRM_Core_Invoke::runItem((Array:15))
#6 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/CRM/Core/Invoke.php(52): CRM_Core_Invoke::_invoke((Array:3))
#7 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/sites/all/modules/civicrm/drupal/civicrm.module(444): CRM_Core_Invoke::invoke((Array:3))
#8 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/includes/menu.inc(527): civicrm_invoke("event", "register")
#9 /home/jenkins/bknix-dfl/build/core-15281-8xhkx/web/index.php(21): menu_execute_active_handler()
#10 {main}
```

File: CRM/Event/Form/Participant.php (hunk 1 above)
Comment (stesi561) [reply CRM/Event/Form/Participant.php]: I assume this change won't clash with custom handling in payment processors. The dummy processor was already saving address data. However if this is a change of behaviour I think it's possibly worth referencing in the PR description?

**Comment by stesi561** [comment]
I've updated original review (r-run and r-explain) .

**Comment by mattwire** [comment]
@stesi561 Thanks for the review!

**Comment by eileenmcnaughton** [comment]
This PR caused https://lab.civicrm.org/dev/event/issues/25

**Pull request merged**